### ISSUE TITLE
Can you only run different model types in the same image?

### ISSUE THREAD

**Issue by kilimchoi** [blib-la/runpod-worker-comfy#51]
Based on readme, it sounds like we have to build a separate docker image for different model type. 
so if we want to support sd2+, sdxl, sd3 models, do we need to build 3 separate docker images?

**Comment by TimPietrusky** [comment]
@kilimchoi thanks for getting in touch!

> so if we want to support sd2+, sdxl, sd3 models, do we need to build 3 separate docker images?

Exactly, this is how it works right now. The idea was, that people can pick one of the images for their use case and use it without the worry about any other model. This makes sure, that the image sizes are "small" and only contain the things that are totally needed for the specific model. 

If you want to have multiple models in one worker, you have two options:

1. You bake them into one image yourself
2. You attach a network-storage and use the `base` image (as it contains no models at all). On the network-storage, you can then add all the models you want.

**Comment by TimPietruskyRunPod** [comment]
Yes — a single image can run multiple model types as long as each one has its required files present (the workflow's \`CheckpointLoaderSimple\` / \`UNETLoader\` / etc. is resolved at request time based on whatever's in the corresponding \`models/<dir>/\`).

Two ways to do this:

1. **Bake everything in:** Add multiple \`wget\` lines in your derived \`Dockerfile\` covering all checkpoints/UNETs/VAEs/CLIPs you need.
2. **Network volume:** Put all model files on a shared network volume under \`models/<dir>/\`. One worker image, many models, no rebuild when you add a new one.

The hub image is FLUX.1-dev-fp8-only on purpose (to keep the default image small), but custom images can absolutely mix.

Closing as answered.
